core/src/main/scala/org/apache/spark/shuffle/ShuffleMemoryManager.scala

```diff
@@ -65,7 +66,7 @@ class ShuffleMemoryManager protected (
    * total memory pool (where N is the # of active tasks) before it is forced to spill. This can
    * happen if the number of tasks increases but an older task had a lot of memory already.
    */
-  def tryToAcquire(numBytes: Long): Long = synchronized {
+  def tryToAcquire(numBytes: Long): Long = memoryManager.synchronized {
```

Comment:
This slightly-confusing synchronization on `memoryManager` isn't a huge deal since it's going to be removed by the memory manager unification patch.